modules/programs/ssh.nix

```diff
@@ -132,6 +139,37 @@ in
       '';
     };
 
+    compression = mkOption {
+      default = false;
+      type = types.bool;
+      description = "Specifies whether to use compression.";
+    };
+
+    serverAliveInterval = mkOption {
+      type = types.int;
+      default = 0;
+      description =
+        "Set default timeout in seconds after which response will be requested.";
+    };
+
+    hashKnownHosts = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Indicates  that ssh(1) should hash host names and addresses when they
+        are added to the known hosts file.
+      '';
+    };
+
+    userKnownHostsFile = mkOption {
+      type = types.str;
```

Comment:
This could be `types.path` but then the default needs to be a Nix string, a `defaultText` needs to be added and `toString` used below when writing `~/.ssh/config`.